File: prolog/p-20.pl
:- ensure_loaded('take-drop').

remove_at(Xs, N, Ys) :-
    N > 0, N1 is N - 1,
    take(N1, Xs, P1),
    drop(N, Xs, P2),
    append(P1, P2, Ys).

File: erlang/p_20.erl
-module(p_20).
-export([remove_at/2]).
-import(take_drop, [take/2, drop/2]).
-import(lists, [append/2]).

remove_at(L, N) when N > 0 ->
    append(take(L, N - 1), drop(L, N)); 
remove_at(_, N) when N < 0 ->
    erlang:error(negative_index).
